### PR TITLE
[Refactor] change the macro style of the InListPredicate 

### PR DESCRIPTION
**Pull request by englefly**
# Proposed changes
refactor in_list_predicate.cpp
1. use normal cpp code instead of macro definition.
2. remvoe NotInListPredicate class. It is almost duplicated to InListPredicate
3. add bool _in_or_not member in InListPredicate. if _in_or_not=false, InListPredicate stands for former NotInListPredicate.

Only rewrite vectorization part. The non-vertorization parts are still in Macros style. remove them when we clean non-verctorization code.

Issue Number: close #xxx

## Problem Summary:

Describe the overview of changes.

## Checklist(Required)

1. Does it affect the original behavior: (Yes/No/I Don't know)
3. Has unit tests been added: (Yes/No/No Need)
6. Has document been added or modified: (Yes/No/No Need)
7. Does it need to update dependencies: (Yes/No)
8. Are there any changes that cannot be rolled back: (Yes/No)

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: be/src/olap/in_list_predicate.h [hunk 1]
```diff
@@ -76,33 +76,55 @@ namespace doris {
...
+                        bool* flags) const override {}
+    void evaluate_or(vectorized::IColumn& column, uint16_t* sel, uint16_t size,
+                        bool* flags) const override {}
+
+private:
+    phmap::flat_hash_set<T> _values;
+    bool _in_or_not; //true for in(...), false for not in (...)
+};
+
+// template <class T>
```
Comment (Gabriel39) [inline be/src/olap/in_list_predicate.h:104]: delete these commented codes?

**Comment by yiguolei** [comment]
very expected work....

**Comment by BiteTheDDDDt** [comment]
Hi, I have done some similar work, maybe you can refer to some designs.
https://github.com/BiteTheDDDDt/incubator-doris/commit/93bdd0582fd6c2890e6f36c1805755712e40553d#diff-1cfbfefc7969508323c47da436eeb1d5788b4ec905166f99b967c248974d9eb1

File: be/src/olap/in_list_predicate.h (hunk 1 above)
Comment (englefly) [reply be/src/olap/in_list_predicate.h:104]: It is better remove these codes when we clean non-vectorized code.

Comment (englefly) [reply be/src/olap/in_list_predicate.h]: The NotInListPredicate definition is removed.

**Pull request closed without merging**